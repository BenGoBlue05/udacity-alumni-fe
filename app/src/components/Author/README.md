## Author Component
A component that ...

### Example

```js
<Author />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
