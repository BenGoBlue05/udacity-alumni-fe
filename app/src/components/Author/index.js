import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Article from 'grommet-udacity/components/Article';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';
import Quote from 'grommet-udacity/components/Quote';

const defaultAvatarUrl = 'https://github.com/RyanCCollins/cdn/blob/master/alumni-webapp/no-user.png?raw=true';

const Author = ({
  author,
}) => (
  <Article className={styles.panel} align="center">
    <Heading tag="h2" align="center">
      About the Author
    </Heading>
    <Box align="center">
      <img src={author.avatar || defaultAvatarUrl} className={styles.avatar} />
    </Box>
    <Heading tag="h3" align="center">
      {author.name}
    </Heading>
    <Quote borderColorIndex="light-2">
      <Heading tag="h5" align="center">
        Preventive care centers of excellence misrepresentation waiting
        period EOB. Second surgical opinion network home health care
        co-insurance NAIC out-of-network.
      </Heading>
    </Quote>
  </Article>
);

Author.propTypes = {
  author: PropTypes.shape({
    name: PropTypes.string.isRequired,
    avatar: PropTypes.string,
  }),
};

export default cssModules(Author, styles);
