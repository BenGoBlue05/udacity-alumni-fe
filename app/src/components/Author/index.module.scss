.panel {
  position: relative;
  width: 100%;
  max-width: 750px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(167, 157, 144, 0.52);
  margin: 80px 0;
  padding: 60px !important;
}

.avatar {
  width: 100px;
  height: 100px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
  padding: 4px;
  line-height: 1.42857143;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 50%;
}
