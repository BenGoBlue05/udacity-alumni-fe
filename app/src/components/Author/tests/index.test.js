import Author from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<Author />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Author
        author={{ name: 'Ryan Collins', avatar: 'ryancollins.io/avarat' }}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
