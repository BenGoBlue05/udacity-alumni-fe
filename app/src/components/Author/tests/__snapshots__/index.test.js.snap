exports[`<Author /> should render with default props 1`] = `<div />`;
